pkg/controller/cassandracluster/reconcile.go

```diff
@@ -120,14 +120,15 @@ func (rcc *ReconcileCassandraCluster) CheckNonAllowedChanged(cc *api.CassandraCl
 		return true
 	}
 
-	if needUpdate, updateStatus := CheckNonAllowedRemoveDC(rcc, cc, status, &oldCRD); needUpdate {
+	var updateStatus string
```

Comment:
@allamand, I can't see why this change is needed ? In all cases we overwrite those 2 variables so why not just keep the old code ?

Reply:
@Orange-cscetbon with the previous change we uses := which in fact creates 2 variables.
so the needUpdate var in this case was not the global needUpdate one which leads to the bug encountered